apps/gateway/lib/gateway/controllers/fallback_controller.ex

```diff
@@ -20,6 +20,22 @@ defmodule Gateway.Controllers.FallbackController do
     send_resp(conn, 400, Jason.encode!(%{"error" => "invalid reward"}))
   end
 
+  def call(conn, {:error, :item_not_found}) do
+    send_resp(conn, 400, Jason.encode!(%{"error" => "item not found"}))
+  end
```

Comment:
I think this case is never happening since the `get_item_by_name/2`  method is returning `{:error, :not_found}`

Reply:
You're wrong it's not for `get_item_by_name/2` it's for `equip_item/3`